### ISSUE TITLE
Missing words in search

### ISSUE THREAD

**Issue by abulhol** [sphinx-doc/sphinx#3039]
This may be related to #2891:
I am using Sphinx 1.5a1 and noticed that some words are completely missing in the search index. 
E.g. I have a sentence that starts with
"Disable System Integrity Protection (SIP)....."

If I search for "system integrity", there is no match, but if I look for "SIP", there is.
I checked the searchindex.js, and the word "integrity" is not there.

I use quite a lot of "include" directives because I use identical text blocks in several places, but I think this should not be an issue here (?). 
(I have added these docs to "exclude_patterns" in conf.py so they do not throw a "document is not included in any toctree")

**Comment by abulhol** [comment]
and btw, there is a (long) URL in the same sentence:
developer.apple.com/library/mac/documentation/Security/ConceptualSystem_Integrity_Protection_Guide/ConfiguringSystemIntegrityProtection/ConfiguringSystemIntegrityProtection.html
and in searchindex.js I find an entry for:
"configuringsystemintegrityprotect"

strange, isn't it?

**Comment by abulhol** [comment]
I now tried out other Sphinx releases (1.4.6, 1.4.8), but the results are the same.
Here are some more strange findings:

If I search for "integrity", I get some results which all do NOT contain the word "integrity", but e.g. "integration". 

Search index looks like:

```
integ : [38, 39],
integr : [0, 1, 28, 36],
integriti : [16, 28],
```

Why does it contain "integriti"? There is no word in my documents that starts with these letters.

**Comment by TimKam** [comment]
Could you create a [minimal, complete and verifiable example](http://stackoverflow.com/help/mcve) that reproduces your issue?

**Comment by tk0miya** [comment]
Thank you for reporting.
The result of stemming is changed for capital words:

```
>>> from sphinx.search.en import SearchEnglish
>>> SearchEnglish(None).stem('Integrity')
'Integriti'
>>> SearchEnglish(None).stem('integrity')
'integr'
```

I think this is a bug of the index builder for English.

**Comment by tk0miya** [comment]
Fixed at e04fe84.

**Comment by abulhol** [comment]
Thanks for fixing this so quickly! :-)
